app/lib/account_session.rb

```diff
@@ -51,13 +51,45 @@ def serialise
 
   def to_hash
     {
-      access_token: access_token,
-      refresh_token: refresh_token,
+      access_token: @access_token,
+      refresh_token: @refresh_token,
       level_of_authentication: level_of_authentication,
     }
   end
 
+  def get_remote_attributes(remote_attributes)
+    remote_attributes.index_with { |name| oidc_do :get_attribute, { attribute: name } }
+  end
+
+  def set_remote_attributes(remote_attributes)
+    return unless remote_attributes.any?
```

Comment:
Just nitpicking but I find `if` much more easy to mentally parse than `unless` so perhaps we can change this to `return if remote_attributes.none?`